tests/benchmarks/micro_benchmarks/test_disk_performance.py

```diff
@@ -0,0 +1,529 @@
+         predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CPU)
+        assert (benchmark_class)
+
+        benchmark = benchmark_class(benchmark_name, parameters='')
+
+        # Check basic information
+        assert (benchmark)
+        ret = benchmark._preprocess()
+        assert (ret is True)
+        assert (benchmark.return_code == ReturnCode.SUCCESS)
+        assert (benchmark.name == 'disk-performance')
+        assert (benchmark.type == BenchmarkType.MICRO)
+
+        # Command list should be empty
+        assert (0 == len(benchmark._commands))
+
+    def test_disk_performance_invalid_block_device(self):
+        """Test disk-performance benchmark command generation with invalid block device."""
+        benchmark_name = 'disk-performance'
+        (benchmark_class,
+         predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CPU)
+        assert (benchmark_class)
+
+        block_devices = ['/dev/null']
+        block_device_option = '--block_devices ' + ' '.join(block_devices)
+
+        benchmark = benchmark_class(benchmark_name, parameters=block_device_option)
+
+        # Check basic information
+        assert (benchmark)
+        ret = benchmark._preprocess()
+        assert (ret is False)
+        assert (benchmark.return_code == ReturnCode.INVALID_ARGUMENT)
+        assert (benchmark.name == 'disk-performance')
+        assert (benchmark.type == BenchmarkType.MICRO)
+
+    def test_disk_performance_benchmark_disabled(self):
+        """Test disk-performance benchmark command generation with all benchmarks disabled."""
+        benchmark_name = 'disk-performance'
+        (benchmark_class,
+         predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CPU)
+        assert (benchmark_class)
+
+        block_devices = ['/dev/nvme0n1', '/dev/nvme1n1']
+        block_device_option = '--block_devices ' + ' '.join(block_devices)
+
+        param_str = block_device_option
+        param_str += ' --enable_seq_precond=0'
+        param_str += ' --rand_precond_time=0'
+        param_str += ' --seq_read_runtime=0'
+        param_str += ' --seq_write_runtime=0'
+        param_str += ' --rand_read_runtime=0'
+        param_str += ' --rand_write_runtime=0'
+        benchmark = benchmark_class(benchmark_name, parameters=param_str)
+
+        # Check basic information
+        assert (benchmark)
+        ret = benchmark._preprocess()
+        assert (ret is True)
+        assert (benchmark.return_code == ReturnCode.SUCCESS)
+        assert (benchmark.name == 'disk-performance')
+        assert (benchmark.type == BenchmarkType.MICRO)
+
+        # Command list should be empty
+        assert (0 == len(benchmark._commands))
+
+    def test_disk_performance_benchmark_enabled(self):
+        """Test disk-performance benchmark command generation with all benchmarks enabled."""
```

Comment:
seems you need to mock `['/dev/nvme0n1', '/dev/nvme1n1']` for unit tests
```suggestion
    from unittest import mock
    @mock.patch('pathlib.Path.is_block_device')
    def test_disk_performance_benchmark_enabled(self):
        """Test disk-performance benchmark command generation with all benchmarks enabled."""
        mock_is_char_device.return_value = True
```

Reply:
Fixed.